### PR TITLE
APIM Product build optimization

### PR DESCRIPTION
**Pull request by mushthaq33**
## Purpose
This PR optimizes the APIM product build

## Goals
To optimize the APIM product build

## Documentation
N/A - Test case optimization

## Security checks
 - Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines? yes
 - Ran FindSecurityBugs plugin and verified report? yes/no
 - Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets? yes
 
## Learning
- http://testng.org/doc/documentation-main.html

### CODE REVIEW

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/am/integration/tests/jwt/JWTTestCase.java [hunk 1]
```diff
@@ -206,7 +202,7 @@ public void testEnableJWTAndClaims() throws Exception {
      * and is hardcoded to 15 mins. So the test case will take approximately 15mins to complete and it will delay the
      * product build unnecessarily, hence the test case is disabled.
      */
-    @Test(groups = { "wso2.am" }, description = "JWT Token generation when JWT caching is enabled", enabled = false)
+    @Test(groups = { "noRestart" }, description = "JWT Token generation when JWT caching is enabled", enabled = false)
```
Comment (uthaiyashankar) [inline modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/am/integration/tests/jwt/JWTTestCase.java]: Why this test is disabled? I can understand the comment saying why we are disabling, but then what is the point of having this test? How do we test the invalidation functionality if we are disabling this test?

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/am/integration/tests/other/APIM710AllSubscriptionsByApplicationTestCase.java [hunk 2]
```diff
@@ -237,7 +222,7 @@ public void setEnvironment() throws Exception{
             Thread.sleep(3000);
```
Comment (uthaiyashankar) [inline modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/am/integration/tests/other/APIM710AllSubscriptionsByApplicationTestCase.java]: This file has several sleeps in between loops. The sleeps are here NOT to wait until getting the response, but rather give delay between sending requests to APIM. APIM should be able to handle multiple requests parallel. So, why do we need sleeps here?

File: modules/integration/tests-integration/tests-backend/src/test/resources/artifacts/AM/configFiles/common/api-manager.xml [hunk 3]
```diff
@@ -0,0 +1,572 @@
+<APIManager>
```
Comment (uthaiyashankar) [inline modules/integration/tests-integration/tests-backend/src/test/resources/artifacts/AM/configFiles/common/api-manager.xml]: Need license header.

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/am/integration/tests/other/APIM710AllSubscriptionsByApplicationTestCase.java (hunk 2 above)
Comment (mushthaq33) [reply modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/am/integration/tests/other/APIM710AllSubscriptionsByApplicationTestCase.java]: Fixed with 4f6054ea9733fd040759435c990d6a53cfb56c7a

File: modules/integration/tests-integration/tests-backend/src/test/resources/artifacts/AM/configFiles/common/api-manager.xml (hunk 3 above)
Comment (mushthaq33) [reply modules/integration/tests-integration/tests-backend/src/test/resources/artifacts/AM/configFiles/common/api-manager.xml]: Fixed with 4f6054ea9733fd040759435c990d6a53cfb56c7a

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/am/integration/tests/jwt/JWTTestCase.java (hunk 1 above)
Comment (mushthaq33) [reply modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/am/integration/tests/jwt/JWTTestCase.java]: This is now enabled and fixed it by reducing the cache timeout in api-manager.xml. Fixed with 4f6054ea9733fd040759435c990d6a53cfb56c7a

**Pull request merged**